credentials/tls.go

```diff
@@ -50,6 +51,22 @@ func (t TLSInfo) AuthType() string {
 	return "tls"
 }
 
+// ValidateAuthority validates that the provided authority being used to
+// override the :authority header is valid by verifying it against the peer
+// certificates. It returns nil on successful validation, or a non-nil error if
```

Comment:
Same nit here. You can just say "It returns a non-nil error if validation fails".